### ISSUE TITLE
new-asbuiltreport : Exception calling "Substring" with "1" argument(s): "startIndex cannot be larger than length of string

### ISSUE THREAD

**Issue by RACranendonk** [AsBuiltReport/AsBuiltReport.VMware.vSphere#68]
**Describe the bug**
Running the new-asbuiltreport results in the following error:

new-asbuiltreport : Exception calling "Substring" with "1" argument(s): "startIndex cannot be larger than length of string.
Parameter name: startIndex"
At line:1 char:1
+ new-asbuiltreport -report VMware.vSphere -username administrator@vsph ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,New-AsBuiltReport

**To Reproduce**
install-module PScribo
install-module asbuiltreport

Run the new-asbuiltreport with the example from the main github page:
PS C:\Windows\system32> new-asbuiltreport -report VMware.vSphere -username administrator@vsphere.local -password <password> -format html -OutputPath 'D:\As-built' -target <vcenter.fqdn>

![image](https://user-images.githubusercontent.com/4656220/64006236-31fc9f00-cb12-11e9-9639-9e88d59a5ea6.png)

**Comment by mattallford** [comment]
Hi @RACranendonk,

Any movement on this one? I did run in to a similar issue where a license had a null value for either the key or the name (I forget which at the moment), so I'm wondering if this issue is the same and it's something we need to account for in the function.

Cheers, Matt.

**Comment by RACranendonk** [comment]
Hi, apologies for the (extremely) delayed response.

I do believe it was due to an empty license field in vCenter.

However, I've just reinstalled AsBuiltReport on the same machine and used it on the same environment, it does work now. I previously ran version 1.0.2, the current release is 1.0.3.

The issue seems to have been resolved! Awesome!

**Comment by tpcarman** [comment]
@RACranendonk I have encountered this issue myself and believe I have a workaround however I am still wanting to know why this issue occurs. Would it be possible for you to connect to your vCenter using PowerCLI and run the following
```
$LicenseObject = @()
$ServiceInstance = Get-View ServiceInstance
$LicenseManager = Get-View $ServiceInstance.Content.LicenseManager
$LicenseManagerAssign = Get-View $LicenseManager.LicenseAssignmentManager
foreach ($License in ($LicenseManager.Licenses <#| Where-Object {$_.licensekey -ne ''}#>)) {
    $Object = @()
    $LicenseExpiration = $License.Properties | Where-Object { $_.Key -eq 'expirationDate' } | Select-Object -ExpandProperty Value
    $Object = [PSCustomObject]@{
        'License' = $License.License                              
        'Product' = $License.Name
        'LicenseKey' = $License.LicenseKey
        'Total' = $License.Total
        'Used' = Switch ($License.Used) {
            $null { "0" }
            default { $License.Used }
        }
        'Expiration' = 
        if ($LicenseExpiration -eq $null) {
            "Never"
        } elseif ($LicenseExpiration -gt (Get-Date)) {
            $LicenseExpiration.ToShortDateString()
        } else {
            "Expired"
        }
    }
    $LicenseObject += $Object
}
Write-Output $LicenseObject | ft
```
Would you be able to post a screenshot of the output as well as a screenshot of your licensing from within the vSphere client? Please obscure your license keys before posting. I'm interested to see which particular licenses are causing the issue.

**Comment by RACranendonk** [comment]
Hi Tim, please see the screenshots below. Again, with the new version I'm not having any issues. It is a VCF environment, that may be relevant in regards to how the licenses are handled.

![image](https://user-images.githubusercontent.com/4656220/70899243-86ab2380-1ff6-11ea-9811-3aca0c0d580b.png)
![image](https://user-images.githubusercontent.com/4656220/70899514-0df89700-1ff7-11ea-9d50-fb3d24d2ceeb.png)

**Comment by tpcarman** [comment]
Thanks @RACranendonk. It looks to have been caused by the vSAN Witness license, which is the same issue I experienced in my home lab. Thanks for testing this, I'll be sure to include the fix in the next update. FYI, I am still trying to find out why the vSAN Witness license does not report properly too.

**Comment by tpcarman** [comment]
Resolved in v1.1.3 [0edcbdf]
